### PR TITLE
Add support for GHC912

### PR DESCRIPTION
**Pull request by albertodvp**
Add support for GHC912.

---
I've prepared this during this event :)
[Open Source Saturday](https://lu.ma/open-source-saturday-torino)

### CODE REVIEW

**Comment by sol** [comment]
Hey @albertodvp 👋

Thanks for looking into this. Please rebase your changes on `main`.

**Review by sol — CHANGES_REQUESTED** [review]
Other than my comment, this looks good 👍

Please make sure to squash all your  changes into a single commit.

File: package.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ library:
     - Test.DocTest.Internal.Cabal
 
   dependencies: &dependencies
-    ghc: ">= 8.0 && < 9.12"
+    ghc: ">= 8.0 && < 9.13"
```
Comment (sol) [inline package.yaml:57]: This was actually correct before. It should be `< 9.14` so that Doctest can be used with development versions of GHC.

**Review by sol — CHANGES_REQUESTED** [review]
Please address that last comment,

File: package.yaml (hunk 1 above)
Comment (sol) [reply package.yaml:57]: I'm gonna take care of it.

**Review by sol — APPROVED** [review]

**Comment by sol** [comment]
Again, thanks a lot for your contribution.

**Comment by albertodvp** [comment]
Thank you for the patience ☺️

**Pull request merged**